packages/@jsii/kernel/src/kernel.ts

```diff
@@ -1325,6 +1304,44 @@ export class Kernel {
     return property;
   }
 
+  /**
+   * Shared (non-public implementation) to as not to break API recording.
+   */
+  private _getBinScriptCommand(
+    req: api.GetScriptCommandRequest,
+  ): api.GetScriptCommandResponse {
+    const packageDir = this._getPackageDir(req.assembly);
+    if (fs.pathExistsSync(packageDir)) {
+      // module exists, verify version
+      const epkg = fs.readJsonSync(path.join(packageDir, 'package.json'));
+
+      if (!epkg.bin) {
+        throw new JsiiFault(
+          'There is no bin scripts defined for this package.',
+        );
+      }
+
+      const scriptPath = epkg.bin[req.script];
+
+      if (!epkg.bin) {
+        throw new JsiiFault(`Script with name ${req.script} was not defined.`);
+      }
+
+      return {
+        command: path.join(packageDir, scriptPath),
+        args: req.args ?? [],
+        env: {
+          ...process.env,
+          // Make sure the current NODE_OPTIONS are honored if we shell out to node
+          NODE_OPTIONS: process.execArgv.join(' '),
```

Comment:
Not everything in `process.execArgv` is guaranteed to be valid to use in `NODE_OPTIONS`... That'll be randomly problematic in certain environments.

Reply:
Same comment as above, this is a direct port of the existing `invokeBinScript` logic.

I think, in this case, the contents of `process.execArgv` is predictable because the JSII runtime spawns the node process. Python is currently the only runtime with bin script support and it appears to only pass `--max-old-space-size=4069`, which shouldn't present a problem:
https://github.com/aws/jsii/blob/main/packages/%40jsii/python-runtime/src/jsii/_kernel/providers/process.py#L252